app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package com.architectcoders.arquitectomarvel.ui.main
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.architectcoders.arquitectomarvel.model.Repository
+import com.architectcoders.arquitectomarvel.model.characters.Result
+import com.architectcoders.arquitectomarvel.model.database.dbItemComics
+import com.architectcoders.arquitectomarvel.model.database.dbObject
+import com.architectcoders.arquitectomarvel.model.database.relations.ResultWithItemsComics
+import com.architectcoders.arquitectomarvel.model.database.relations.toListResult
+import com.architectcoders.arquitectomarvel.ui.common.Event
+import com.architectcoders.arquitectomarvel.ui.common.Scope
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import java.net.UnknownHostException
+
+class MainViewModel(private val repository: Repository) : ViewModel(), Scope by Scope.Impl() {
+
+    sealed class UiModel {
+        object Loading : UiModel()
+        class GetRemoteData(val results: List<Result>) : UiModel()
+        class GetErrorMessage(val message: String) : UiModel()
+    }
+
+    private val dao by lazy { repository.dao }
+
+    private val _model = MutableLiveData<UiModel>()
+    val model: LiveData<UiModel>
+        get() {
+            if (_model.value == null) refresh()
+            return _model
+        }
+
+    private val _navigation = MutableLiveData<Event<Result>>()
+    val navigation: LiveData<Event<Result>> = _navigation
+    init {
+        initScope()
+    }
+
+    private fun refresh() {
+
+
+        launch {
+            try {
+                _model.value = UiModel.Loading
+                val characters = repository.getCharactersRemote()
+                val resultsRemote = characters.characterData?.results ?: emptyList()
+                _model.value = UiModel.GetRemoteData(resultsRemote)
+
+                resultsRemote.forEach { result ->
+                    dao.insertResult(result.dbObject)
+                    val colectionUri = result.comics.collectionURI
+                    result.comics.items.forEach { item ->
+                        dao.insertComics(item.dbItemComics(colectionUri))
+                    }
+                }
+
```

Comment:
@jujoru I was wondering if we have many spaces between blocks (like lines 42, 43, 58, and 59). What do you think?

Reply:
Right, I have to get used to formatting the code before doing push. I review it.